### PR TITLE
Fixed misc CRAN issues

### PR DESCRIPTION
**Pull request by jameslamb**
Hello,

Thank you for this excellent package! @randxie and I would like to help you address #629 . We have some experience on the team working with the wonderful eccentricities of getting accepted to CRAN.

**What I changed**

Please consider this first PR to address some low-hanging notes and warnings. I've also attached a file with the state of the `R CMD CHECK` output showing the remaining notes and warnings that could be addressed in follow-up PRs.

[r_cmd_check.log](https://github.com/Microsoft/LightGBM/files/1805559/r_cmd_check.log)

As of this PR, R CMD CHECK for this package is down to 4 warnings and 3 notes.

**Setup Information**

I built the package by running the following from the repo root (I am on MacOS):

```
# Avoid errors related to AppleClang
export CXX=g++-7 CC=gcc-7

# Build some artifacts and document
Rscript -e "devtools::document()"

# Build the package tarball
cd R-package && Rscript build_package.R

# re-running documentation
Rscript -e "devtools::document()"

# Run R CMD CHECK and save results to file
R CMD CHECK lightgbm_2.1.0.tar.gz --as-cran --no-tests | tee r_cmd_check.log | cat
```

Please let me know if you have any questions or concerns! Please also let @randxie and I know if you are open to us making additional contributions to move this package toward CRAN-readiness.

Thanks!

-James

### CODE REVIEW

File: R-package/DESCRIPTION [hunk 1]
```diff
@@ -5,7 +5,7 @@ Version: 2.1.0
 Date: 2018-01-25
 Author: Guolin Ke <guolin.ke@microsoft.com>
 Maintainer: Guolin Ke <guolin.ke@microsoft.com>
-Description: LightGBM is a gradient boosting framework that uses tree based learning algorithms. 
+Description: Tree based algorithms can be improved by introducing boosting frameworks. LightGBM is one such framework.
```
Comment (jameslamb) [inline R-package/DESCRIPTION]: This is in response to my least-favorite CRAN note. Your package's Description cannot start with the words "this package" or the name of the package.

File: R-package/R/lgb.cv.R [hunk 2]
```diff
@@ -16,10 +16,8 @@ CVBooster <- R6Class(
   )
 )
 
-#' Main CV logic for LightGBM
-#'
-#' Main CV logic for LightGBM
-#'
+#' @title Main CV logic for LightGBM
+#' @name lgb.cv
```
Comment (jameslamb) [inline R-package/R/lgb.cv.R]: this and `lgb.train` were getting grouped into a single documentation object with interwoven fields and R CMD CHECK complained about duplicated parameters. Using explicit `@name` elements avoids this

File: R-package/R/lightgbm.R [hunk 3]
```diff
@@ -122,5 +122,28 @@ NULL
 # Various imports
 #' @import methods
 #' @importFrom R6 R6Class
-#' @useDynLib lightgbm
+#' @useDynLib lib_lightgbm
```
Comment (jameslamb) [inline R-package/R/lightgbm.R]: I was unable to build the package without making this change

File: R-package/R/lightgbm.R [hunk 4]
```diff
@@ -122,5 +122,28 @@ NULL
...
 NULL
+
+# Suppress false positive warnings from R CMD CHECK about
+# "unrecognized global variable"
+globalVariables(c(
```
Comment (jameslamb) [inline R-package/R/lightgbm.R]: This suppresses the `R CMD CHECK` warning about "unrecognized global variables"

File: R-package/src/Makevars [hunk 5]
```diff
@@ -0,0 +1 @@
+
```
Comment (jameslamb) [inline R-package/src/Makevars]: these address warnings from CRAN about having `Make` scripts with only LF characters in them

**Comment by guolinke** [comment]
ping @Laurae2

**Comment by Laurae2** [comment]
@guolinke Please review the R-package description if you need to make changes on it.

You can compare with xgboost: https://github.com/dmlc/xgboost/blob/master/R-package/DESCRIPTION

**Comment by chivee** [comment]
@guolinke , adding @jameslamb and @Laurae2  to the author list will be nice.

@jameslamb , could you please also sign the CLA?

File: R-package/DESCRIPTION (hunk 1 above)
Comment (guolinke) [reply R-package/DESCRIPTION]: @jameslamb I think this Description is too short. We can copy some from https://github.com/Microsoft/LightGBM/blob/master/README.md

Comment (jameslamb) [reply R-package/DESCRIPTION]: @guolinke would you like me to make that change on this PR?

**Comment by jameslamb** [comment]
@chivee sorry for the delay. CLA signed!

File: R-package/DESCRIPTION (hunk 1 above)
Comment (guolinke) [reply R-package/DESCRIPTION]: yeah

File: R-package/DESCRIPTION [hunk 6]
```diff
@@ -5,13 +5,16 @@ Version: 2.1.0
...
+Description: Tree based algorithms can be improved by introducing boosting frameworks. LightGBM is one such framework, and this package offers an R interface to work with it.
     It is designed to be distributed and efficient with the following advantages:
         1. Faster training speed and higher efficiency.
         2. Lower memory usage.
         3. Better accuracy.
         4. Parallel learning supported.
         5. Capable of handling large-scale data.
+    In recognition of these advantages, LightGBM has being widely-used in many winning solutions of machine learning competitions.
+
+    Comparison experiments on public datasets suggest that LightGBM can outperform existing boosting frameworks on both efficiency and accuracy, with significantly lower memory consumption. In addition, parallel experiments suggest that in certain circumstances, LightGBM can achieve a linear speed-up in training time by using multiple machines.
```
Comment (jameslamb) [inline R-package/DESCRIPTION]: @guolinke per your request, I've included additional details here. You will recognize most of this from the README you linked me too. I also added the line about "this is the R interface to the project" following the way XGBoost structured their DESCRIPTION: https://cran.r-project.org/web/packages/xgboost/index.html

I think that's a good clarification for people, to say that `LightGBM` is a framework and this is the interface to it implemented in a particular programming language.

**Comment by guolinke** [comment]
@Laurae2 can you create another PR to add yourself and other R's contributors to author List ?

**Comment by Laurae2** [comment]
@guolinke Yes I'll do when I get time tonight or next week.

Do I add both @jameslamb and @yanyachen ?

**Comment by Jorgfevi** [comment]
Hola , que tal amigos 
Tengo el archivo .exe y el dll , pero a la hora de ejecutar o tratar de instalar la libreria  R que dice build_r.R me sale un error como este 
installation of package ‘C:/Users/BP2499/Documents/R/R-3.5.2/LightGbm/lightgbm_2.2.3.tar.gz’ had non-zero exit status
Desearia saber a que se debe este problema , o que pasos se siguen despues de tener el archivo .exe y .dll 
Gracias

**Pull request merged**